src/components/plots/GeneSearchBar.jsx

```diff
@@ -25,31 +26,56 @@ const GeneSearchBar = (props) => {
 
   const [options, setOptions] = useState([]);
 
-  // pass reactive component as value (search text) to allow auto clear on select
   const [value, setValue] = useState('');
 
   const onOptionSelect = (newGene) => {
-    const genes = [...config?.selectedGenes, newGene];
-
-    onSelect(genes);
-    setValue('');
+    const genes = value.split(/(?<!-)[,\s]+(?!-)/);
+    genes.splice(-1, 1, `${newGene}, `);
+    setValue(genes.join(', '));
+    setOptions([]);
   };
 
-  const onSearch = (searchText) => {
-    setValue(searchText);
+  const onSearch = (input) => {
+    setValue(input);
+
+    const genes = input.split(/(?<!-)[,\s]+(?!-)/);
+    const searchText = genes[genes.length - 1];
+
     setOptions(!searchText ? [] : filterGenes(searchText, geneList, config?.selectedGenes));
   };
 
+  const onClick = () => {
+    if (value === '') return;
+
+    const newGenes = value.split(/(?<!-)[,\s]+(?!-)/).map((gene) => gene.trim()).filter((gene) => geneList.includes(gene));
+    const genes = _.uniq([...config?.selectedGenes, ...newGenes]);
+
+    if (_.isEqual(genes, config?.selectedGenes)) return;
```

Comment:
This piece of code is to handle the case where the users selects a gene that's already selected right? Instead of doing this, would it be simpler to remove genes that are already selected from the list of genes that is displayed?

Reply:
The genes are already disabled in the gene search if they have already been added. This doesn't prevent attepting to add them with the add button, which could happen if:
- the entire name is written manually instead of selecting an option and then added
- a pasted list of genes includes genes that are already loaded